### PR TITLE
RDK-39259: Gamepad rdkplugin added

### PR DESCRIPTION
**Pull request by mateusz-hobgarski-red**
### Description
Gamepad rdkplugin added. This plugin sets up everything so that game pad input events could be received by containerized app.
### Test Procedure
Run containerized app that uses gamepad. Check if it receives game pad events.

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (doesn't fit into the above categories - e.g. documentation updates)

### Requires Bitbake Recipe changes?
- [ ] The base Bitbake recipe (`meta-rdk-ext/recipes-containers/dobby/dobby.bb`) must be modified to support the changes in this PR (beyond updating `SRC_REV`)

### CODE REVIEW

File: rdkPlugins/Gamepad/source/GamepadPlugin.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    return dependencies;
+}
+
+// Begin private methods
+
+int GamepadPlugin::getInputGroupId() const
+{
+    std::ifstream etcGroupFile;
+
+    etcGroupFile.open("/etc/group");
```
Comment (Koczek9) [inline rdkPlugins/Gamepad/source/GamepadPlugin.cpp:118]: We could use getgrname instead https://linux.die.net/man/3/getgrnam

File: CMakeLists.txt [hunk 2]
```diff
@@ -298,6 +298,8 @@ option(PLUGIN_APPSERVICES "Include AppServices plugin" OFF)
 option(PLUGIN_IONMEMORY "Include ION Memory plugin" OFF)
 option(PLUGIN_DEVICEMAPPER "Include DeviceMapper plugin" OFF)
 
+option(PLUGIN_GAMEPAD "Include GamepadPlugin plugin" ON)
```
Comment (Koczek9) [inline CMakeLists.txt:301]: Should be default OFF

File: rdkPlugins/Gamepad/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Dobby RDK GPU Memory Plugin
```
Comment (Koczek9) [inline rdkPlugins/Gamepad/README.md:1]: This whole readme file is incorrect. Make sure that it describes your plugin.

File: rdkPlugins/Gamepad/source/GamepadPlugin.cpp [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "GamepadPlugin.h"
+
+// #include <linux/limits.h>
```
Comment (Koczek9) [inline rdkPlugins/Gamepad/source/GamepadPlugin.cpp:22]: Remove unused libraries

File: rdkPlugins/Gamepad/source/GamepadPlugin.cpp [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+/**
+ * @brief Dobby Hook - run in host namespace *once* when container bundle is downloaded
+ */
+bool GamepadPlugin::postInstallation()
+{
+    AI_LOG_FN_ENTRY();
+
+    // 1. add devices
+    rt_config_linux_resources* schemaResources = mContainerConfig->linux->resources;
```
Comment (Koczek9) [inline rdkPlugins/Gamepad/source/GamepadPlugin.cpp:60]: Please create add_controller_device (or similar) function and use it to separate the logic

File: rdkPlugins/Gamepad/source/GamepadPlugin.cpp [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+
+    // 2. add /dev/input mount, "nodev"(MS_NODEV) flag removed
+    mUtils->addMount("/dev/input/", "/dev/input/", "bind", {"bind", "nosuid", "noexec"});
+
+
+    // 3. add input group to gidMapping
+    int inputGroupId = getInputGroupId();
+    //printf("inputGroupId: %d\n", inputGroupId);
+    rt_defs_id_mapping* newMapping = (rt_defs_id_mapping*)calloc(1, sizeof(rt_defs_id_mapping));
```
Comment (Koczek9) [inline rdkPlugins/Gamepad/source/GamepadPlugin.cpp:94]: Please create add_group_id (or similar) function and use it to separate the logic

File: rdkPlugins/Gamepad/source/GamepadPlugin.cpp [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+    schemaResources->devices = (rt_defs_linux_device_cgroup**)realloc(schemaResources->devices, sizeof(rt_defs_linux_device_cgroup*) * schemaResources->devices_len);
+
+    rt_defs_linux_device_cgroup* device = schemaResources->devices[old_devices_len];
+
+    for (int i = 0; i < NUM_CONTROLLERS; ++i)
+    {
+        device = (rt_defs_linux_device_cgroup*)calloc(1, sizeof(rt_defs_linux_device_cgroup));
+        device->type = strdup("c");
+        device->access = strdup("rw");
+        device->major = 13;
```
Comment (goruklu) [inline rdkPlugins/Gamepad/source/GamepadPlugin.cpp:75]: please define and use DEV_INPUT_EVENT_MAJOR=13

**Review by goruklu — CHANGES_REQUESTED** [review]

File: rdkPlugins/Gamepad/README.md (hunk 3 above)
Comment (mateusz-hobgarski-red) [reply rdkPlugins/Gamepad/README.md]: Corrected

File: rdkPlugins/Gamepad/source/GamepadPlugin.cpp (hunk 7 above)
Comment (mateusz-hobgarski-red) [reply rdkPlugins/Gamepad/source/GamepadPlugin.cpp]: Corrected

File: CMakeLists.txt (hunk 2 above)
Comment (mateusz-hobgarski-red) [reply CMakeLists.txt]: Corrected

File: rdkPlugins/Gamepad/source/GamepadPlugin.cpp (hunk 5 above)
Comment (mateusz-hobgarski-red) [reply rdkPlugins/Gamepad/source/GamepadPlugin.cpp:111]: Done

File: rdkPlugins/Gamepad/source/GamepadPlugin.cpp (hunk 6 above)
Comment (mateusz-hobgarski-red) [reply rdkPlugins/Gamepad/source/GamepadPlugin.cpp:145]: Done

File: rdkPlugins/Gamepad/source/GamepadPlugin.cpp (hunk 1 above)
Comment (mateusz-hobgarski-red) [reply rdkPlugins/Gamepad/source/GamepadPlugin.cpp:186]: I would leave it as it is. We only need to work with /etc/group file and that's already implemented.

**Review by Koczek9 — APPROVED** [review]
LGTM

**Review by goruklu — APPROVED** [review]

**Pull request merged**